File: src/main/java/gg/mineral/practice/commands/settings/SettingsCommand.java
package gg.mineral.practice.commands.settings;

import gg.mineral.practice.commands.PlayerCommand;
import gg.mineral.practice.inventory.menus.SettingsMenu;
import gg.mineral.practice.managers.ProfileManager;

public class SettingsCommand extends PlayerCommand {

	public SettingsCommand() {
		super("settings");
	}

	@Override
	public void execute(org.bukkit.entity.Player player, String[] args) {
		ProfileManager.getOrCreateProfile(player).openMenu(new SettingsMenu());
	}
}

File: src/main/kotlin/gg/mineral/practice/commands/settings/SettingsCommand.kt
package gg.mineral.practice.commands.settings

import dev.rollczi.litecommands.annotations.command.Command
import dev.rollczi.litecommands.annotations.context.Context
import dev.rollczi.litecommands.annotations.execute.Execute
import gg.mineral.practice.entity.Profile
import gg.mineral.practice.inventory.menus.SettingsMenu

@Command(name = "settings")
class SettingsCommand {
    @Execute
    fun execute(@Context profile: Profile) = profile.openMenu(SettingsMenu())
}
